File: Spark-Learning/src/scala/com/atguigu/sparkcore/examples/broadcast/Spark_broadcast.scala
package com.atguigu.sparkcore.examples.broadcast

import org.apache.spark.broadcast.Broadcast
import org.apache.spark.{SparkConf, SparkContext}

object Spark_broadcast {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
    conf.setMaster("local")
    conf.setAppName("broadCast")

    val sc = new SparkContext(conf)
    val blackList = List[String]("hello spark3", "hello spark4")

    val bc: Broadcast[List[String]] = sc.broadcast(blackList)
    val lines = sc.textFile("./data/words")


    lines.filter(line => {
      blackList.contains(line)
      //       val value: List[String] = bc.value
      //      value.contains(line)
    }).foreach(println)
  }
}

File: Spark-Learning/src/java/com/atguigu/spark/core/examples/Spark_broadcast.java
package com.atguigu.spark.core.examples;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.api.java.function.VoidFunction;
import org.apache.spark.broadcast.Broadcast;

import java.util.ArrayList;
import java.util.List;

/**
 * Spark广播
 */
public class Spark_broadcast {
    public static void main(String[] args) {
        SparkConf conf  = new SparkConf();
        conf.setMaster("local");
        conf.setAppName("broadcast");
        JavaSparkContext sc = new JavaSparkContext(conf);
        JavaRDD<String> lines = sc.textFile("./data/words");

        List<String> list = new ArrayList<String>();
        list.add("hello spark3");
        list.add("hello spark4");

        Broadcast<List<String>> bc = sc.broadcast(list);

        lines.filter(new Function<String, Boolean>() {
            @Override
            public Boolean call(String v1) throws Exception {
                return bc.value().contains(v1);
            }
        }).foreach(new VoidFunction<String>() {
            @Override
            public void call(String s) throws Exception {
                System.out.println(s);
            }
        });


    }
}
